docs/source/notebooks/AutoEncoders.md

```diff
@@ -73,99 +75,91 @@ print(x_test.shape)
+    
+    def _keras_build_fn(self, encoding_dim: int, meta: Dict[str, Any]):
+        n_features_in = meta["n_features_in_"]
 
-def get_inverse_tf_model(fit_model: keras.Model, encoding_dim: int) -> keras.Model:
-    """Get an deencoder model.
+        encoder_input = keras.Input(shape=(n_features_in,))
+        encoder_output = keras.layers.Dense(encoding_dim, activation='relu')(encoder_input)
+        encoder_model = keras.Model(encoder_input, encoder_output)
 
-    We do this by extracting the deencoding layer from the fitted autoencoder model
-    and adding a new Keras input layer.
-    """
-    encoded_input = keras.Input(shape=(encoding_dim,))
-    output = fit_model.get_layer("output")(encoded_input)
-    return keras.Model(encoded_input, output)
-```
+        decoder_input = keras.Input(shape=(encoding_dim,))
+        decoder_output = keras.layers.Dense(n_features_in, activation='sigmoid', name="decoder")(decoder_input)
+        decoder_model = keras.Model(decoder_input, decoder_output)
+        
+        autoencoder_input = keras.Input(shape=(n_features_in,))
+        encoded_img = encoder_model(autoencoder_input)
+        reconstructed_img = decoder_model(encoded_img)
 
-Next we create a class that that will enable the `transform` and `fit_transform` methods, as well as integrating all three of our models into a single estimator.
+        autoencoder_model = keras.Model(autoencoder_input, reconstructed_img)
 
-```python
-from sklearn.base import TransformerMixin, clone
-from scikeras.wrappers import BaseWrapper
-
-
-class KerasTransformer(BaseWrapper, TransformerMixin):
-    """A class that enables transform and fit_transform.
-    """
-
-    def __init__(self, *args, tf_est: BaseWrapper = None, inv_tf_est: BaseWrapper = None, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self.tf_est = tf_est
-        self.inv_tf_est = inv_tf_est
+        self.encoder_model_ = BaseWrapper(encoder_model, predict__verbose=self.predict__verbose)
+        self.decoder_model_ = BaseWrapper(decoder_model, predict__verbose=self.predict__verbose)
 
+        return autoencoder_model
 
     def fit(self, X, sample_weight=None):
         super().fit(X=X, y=X, sample_weight=sample_weight)
-        self.tf_est_ = clone(self.tf_est)
-        self.inv_tf_est_ = clone(self.inv_tf_est)
-        self.tf_est_.set_params(fit_model=self.model_)
-        self.inv_tf_est_.set_params(fit_model=self.model_, encoding_dim=self.encoding_dim)
-        X = self.feature_encoder_.transform(X)
-        self.tf_est_.initialize(X=X)
-        X_tf = self.tf_est_.predict(X=X)
-        self.inv_tf_est_.initialize(X_tf)
+        X_tf = self.encoder_model_.initialize(X).predict(X)
+        self.decoder_model_.initialize(X_tf)
         return self
+    
+    def score(self, X) -> float:
+        reconstructed = self.predict(X)
+        return binary_accuracy(X, reconstructed).numpy().mean()
```

Comment:
What type are `X` and `reconstructed`? It appears that have to be TF arrays.

Why can't you do `return (self.predict(X) == X).mean()`?

Reply:
It could be made public. But generally I think of making a method public when it needs to be called from outside the class. This only ever called from `fit`, so to me it makes sense to keep it private.

Reply:
Wrong thread for the previous comment.

As far as I can tell, they are Numpy arrays. TF metrics/losses usually also accept Numpy inputs.

> Why can't you do return (self.predict(X) == X).mean()?

More than likely that we can do that, I was probably just being a bonehead.

Reply:
It'd be really useful to add type hints. I asked because calling `.numpy` on a NumPy array raises an exception.

Reply:
See #201.

Reply:
Thanks for opening #201.

> Why can't you do return (self.predict(X) == X).mean()?

We could. I made a dumb mistake though and put in `binary_accuracy`, I meant to use `binary_crossentropy`. But really it would be better to use `MSE`, since that is an actual metric. I pushed a commit to do just that. Well, actually we use `1-MSE` because sklearn always maximizes `score` in `GridSearchCV` and such. I added a comment to to clarify that in the code.

Reply:
I think the type-hint can be continued in #201.
For this PR, I added some type hints in the notebook for the return type from `Model.predict`. That that coupled with the removal of the code that originally caused the confusion should be enough for this PR.